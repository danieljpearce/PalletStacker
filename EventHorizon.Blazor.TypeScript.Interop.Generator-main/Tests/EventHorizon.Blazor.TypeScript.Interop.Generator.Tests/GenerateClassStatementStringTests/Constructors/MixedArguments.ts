declare module Examples {
    export class ThinEngine { }
    export class Effect { }
    export interface IEffectCreationOptions { }
    export interface IEffectFallbacks { }

    export class ExampleClass {
        constructor(baseName: any,
            attributesNamesOrOptions: string[] | IEffectCreationOptions,
            uniformsNamesOrEngine: string[] | ThinEngine,
            samplers?: Nullable<string[]>,
            engine?: ThinEngine,
            defines?: Nullable<string>,
            fallbacks?: Nullable<IEffectFallbacks>,
            onCompiled?: Nullable<(effect: Effect) => void>,
            onError?: Nullable<(effect: Effect, errors: string) => void>,
            indexParameters?: any);

    }
}