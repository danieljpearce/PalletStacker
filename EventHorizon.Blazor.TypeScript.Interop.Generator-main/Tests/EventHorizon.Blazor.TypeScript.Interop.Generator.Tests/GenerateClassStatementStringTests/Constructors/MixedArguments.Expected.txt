/// Generated - Do Not Edit
namespace Examples
{
    using System;
    using System.Collections.Generic;
    using System.Text.Json.Serialization;
    using System.Threading.Tasks;
    using EventHorizon.Blazor.Interop;
    using EventHorizon.Blazor.Interop.Callbacks;
    using EventHorizon.Blazor.Interop.ResultCallbacks;
    using Microsoft.JSInterop;

    
    
    [JsonConverter(typeof(CachedEntityConverter<ExampleClass>))]
    public class ExampleClass : CachedEntityObject
    {
        #region Static Accessors

        #endregion

        #region Static Properties

        #endregion

        #region Static Methods

        #endregion

        #region Accessors

        #endregion

        #region Properties

        #endregion
        
        #region Constructor
        public ExampleClass() : base() { }

        public ExampleClass(
            ICachedEntity entity
        ) : base(entity)
        {
            ___guid = entity.___guid;
        }

        public ExampleClass(
            object baseName, string[] attributesNamesOrOptions, string[] uniformsNamesOrEngine, string[] samplers = null, ThinEngine engine = null, string defines = null, IEffectFallbacks fallbacks = null, ActionCallback<Effect> onCompiled = null, ActionCallback<Effect, string> onError = null, object indexParameters = null
        )
        {
            var entity = EventHorizonBlazorInterop.New(
                new string[] { "Examples", "ExampleClass" },
                baseName, attributesNamesOrOptions, uniformsNamesOrEngine, samplers, engine, defines, fallbacks, onCompiled, onError, indexParameters
            );
            ___guid = entity.___guid;
        }
        #endregion

        #region Methods

        #endregion
    }
}